resources/domain/signer/index.ts

```diff
@@ -1,9 +1,33 @@
-import Signer from '../../../main/signers/Signer'
+import type Signer from '../../../main/signers/Signer'
 
-export function getSignerType (signer: Signer) {
-  return ['ring', 'seed'].includes((signer.type || '').toLowerCase()) ? 'hot' : signer.type
+// in order of increasing priority
+export enum Type {
+  Ring = 'ring',
+  Seed = 'seed',
+  Aragon = 'aragon',
+  Trezor = 'trezor',
+  Ledger = 'ledger',
+  Lattice = 'lattice'
 }
 
-export function isHardwareSigner (type = '') {
+export function getSignerType (typeValue: string) {
+  return Object.values(Type).find(type => type === typeValue)
+}
+
+export function getSignerDisplayType (signer: Signer) {
+  return ['ring', 'seed'].includes(signer.type.toLowerCase()) ? 'hot' : signer.type
+}
+
+export function isHardwareSignerType (type = '') {
```

Comment:
You can make this function handle both a signer object and a type, then rename back to `isHardwareSigner` which improves readability.